frontend/src/components/AuditLog.vue

```diff
@@ -0,0 +1,254 @@
+                          {{ t('common.next') }}
+                        </button>
+                      </div>
+                    </nav>
+                  </td>
+                </tr>
+              </tfoot>
+            </table>
+          </div>
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { Popover, PopoverButton, PopoverGroup, PopoverPanel } from '@headlessui/vue';
+import { ChevronDownIcon } from '@heroicons/vue/20/solid';
+import { computed, onMounted, ref } from 'vue';
+import { useI18n } from 'vue-i18n';
+import backend, { AuditEventDto } from '../common/backend';
+import FetchError from './FetchError.vue';
+
+const { t, d } = useI18n({ useScope: 'global' });
+
+const auditEvents = ref<AuditEventDto[]>();
+const onFetchError = ref<Error | null>();
+
+const startDate = ref(beginOfDate(new Date()));
+const startDateFilter = ref(startDate.value.toISOString().split('T')[0]);
+const endDate = ref(endOfDate(new Date()));
+const endDateFilter = ref(endDate.value.toISOString().split('T')[0]);
+
+const filterIsReset = computed(() => {
+  return startDateFilter.value == startDate.value.toISOString().split('T')[0]
+    && endDateFilter.value == endDate.value.toISOString().split('T')[0];
+});
+const startDateFilterIsValid = computed(() => {
+  if (!/^\d{4}-\d{2}-\d{2}$/.test(startDateFilter.value)) {
+    return false;
+  }
+  const startDate = new Date(startDateFilter.value);
+  return !isNaN(startDate.getTime());
+});
+const endDateFilterIsValid = computed(() => {
+  if (!/^\d{4}-\d{2}-\d{2}$/.test(endDateFilter.value)) {
+    return false;
+  }
+  const endDate = new Date(endDateFilter.value);
+  if (isNaN(endDate.getTime())) {
+    return false;
+  }
+  if (startDateFilterIsValid.value) {
+    const startDate = new Date(startDateFilter.value);
+    return startDate <= endDate;
+  }
+  return true;
+});
+const filterIsValid = computed(() => {
+  return startDateFilterIsValid.value && endDateFilterIsValid.value;
+});
+
+const currentPage = ref(1);
+const pageSize = ref(20);
+const paginationBegin = computed(() => {
+  if (auditEvents.value) {
+    return (currentPage.value - 1) * pageSize.value + Math.min(1, auditEvents.value.length);
+  } else {
+    return 0;
+  }
+});
+const paginationEnd = computed(() => {
+  if (auditEvents.value) {
+    return (currentPage.value - 1) * pageSize.value + auditEvents.value.length;
+  } else {
+    return 0;
+  }
+});
+const hasNextPage = ref(false);
+var pages = [0];
+
+onMounted(fetchData);
+
+async function fetchData() {
+  onFetchError.value = null;
+  try {
+    const events = await backend.auditLogs.getAllEvents(startDate.value, endDate.value, pages[currentPage.value - 1], pageSize.value + 1);
+    if (currentPage.value == 1 && pages[0] == 0 && events.length > 0) {
+      pages[0] = events[0].id - 1;
+    }
```

Comment:
this is really hard to read/understand.

Reply:
Fixed in d7fff0aa8bd4108d09dbfacb14315161dd90108f